### ISSUE TITLE
API: publish errors (POST /v3/artifacts/collections/) are not in v3 error style (list instead of object)

### ISSUE THREAD

**Issue by alikins** [ansible/galaxy-dev#128]
A publish request that POST a collection artifact (POST http://localhost:5001/api/automation-hub/v3/artifacts/collections/) that already exists returns a 400 error with a json body that is a list with one error string, contrary to the v3 error spec (https://petstore.swagger.io/?url=https://raw.githubusercontent.com/ansible/galaxy-api/master/openapi/openapi.yaml)

Request:

```
POST http://localhost:5001/api/automation-hub/v3/artifacts/collections/ HTTP/1.1
```

Request headers:
```
Accept-Encoding:	identity
Host:	localhost:5001
User-Agent:	Python-urllib/3.6
Content-Type:	multipart/form-data; boundary=--------------------------1224132cd91e497b8a2e3e2b8d6acec1
Content-Length:	28314
Connection:	close
x-rh-identity:	<TOKEN REDACTED FOR BREVITY>
```

Request body:
```
----------------------------1224132cd91e497b8a2e3e2b8d6acec1
Content-Disposition: form-data; name="sha256"
c45e46694f5d6f8cab3138aa1e1eac98470add36
----------------------------1224132cd91e497b8a2e3e2b8d6acec1
Content-Disposition: file; name="file"; filename="alikins-collection_inspect-0.0.62.tar.gz"
Content-Type: application/octet-stream
<OCTET STREAM REDACTED FOR BREVITY>
```

Response headers:

```
HTTP/1.1 400 Bad Request
Date:	Tue, 15 Oct 2019 13:00:29 GMT
Server:	WSGIServer/0.2 CPython/3.6.8
Content-Type:	application/json
Vary:	Accept
Allow:	POST, OPTIONS
X-Frame-Options:	SAMEORIGIN
Content-Length:	28
```

Response body:
``` json
[
    "Artifact already exists."
]
```

The expected form would be something like:

``` json
 {
   "errors": [
    {
      "code": "artifact_exists",
      "title": "Artifact already exists",
      "status": "404",
      }
  ]
}
```

**Comment by alikins** [comment]
Not sure what is expected to be returned by pulp, there are no error responses documented for https://pulp-ansible.readthedocs.io/en/latest/restapi.html#tag/ansible:-collections

Either way, galaxy-api needs to return the errors in galaxy v3 style.

**Comment by alikins** [comment]
Example of this cause ansible-galaxy to traceback:

```
(ansible-py3) [newswoop:F29:ansible (stable-2.9 % u=)]$ HTTP_PROXY=127.0.0.1:8080 HTTPS_PROXY=127.0.0.1:8080 ANSIBLE_GALAXY_IGNORE=true ANSIBLE_CONFIG=ansible-ah-rhid-local.cfg ansible-galaxy collection publish -s local_hub  /home/adrian/src/testcollections/collection_inspect/releases/alikins-collection_inspect-0.0.62.tar.gz -vvvvv
ansible-galaxy 2.9.0rc3.post0
  config file = /home/adrian/src/ansible/ansible-ah-rhid-local.cfg
  configured module search path = ['/home/adrian/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/adrian/src/ansible/lib/ansible
  executable location = /home/adrian/src/ansible/bin/ansible-galaxy
  python version = 3.6.9 (default, Jul  3 2019, 17:57:57) [GCC 8.3.1 20190223 (Red Hat 8.3.1-2)]
Using /home/adrian/src/ansible/ansible-ah-rhid-local.cfg as config file
Initial connection to galaxy_server: http://localhost:5001/api/automation-hub/
Opened /home/adrian/.ansible/galaxy_token
Calling Galaxy at http://localhost:5001/api/automation-hub/
Found API version 'v3' with Galaxy server local_hub (http://localhost:5001/api/automation-hub/)
Publishing collection artifact '/home/adrian/src/testcollections/collection_inspect/releases/alikins-collection_inspect-0.0.62.tar.gz' to local_hub http://localhost:5001/api/automation-hub/
Calling Galaxy at http://localhost:5001/api/automation-hub/v3/artifacts/collections/
ERROR! Unexpected Exception, this is probably a bug: 'list' object has no attribute 'get'
the full traceback was:

Traceback (most recent call last):
  File "/home/adrian/src/ansible/lib/ansible/galaxy/api.py", line 181, in _call_galaxy
    method=method, timeout=20)
  File "/home/adrian/src/ansible/lib/ansible/module_utils/urls.py", line 1390, in open_url
    unredirected_headers=unredirected_headers)
  File "/home/adrian/src/ansible/lib/ansible/module_utils/urls.py", line 1294, in open
    r = urllib_request.urlopen(*urlopen_args)
  File "/usr/lib64/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib64/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/usr/lib64/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib64/python3.6/urllib/request.py", line 570, in error
    return self._call_chain(*args)
  File "/usr/lib64/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/lib64/python3.6/urllib/request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: Bad Request

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/adrian/src/ansible/bin/ansible-galaxy", line 123, in <module>
    exit_code = cli.run()
  File "/home/adrian/src/ansible/lib/ansible/cli/galaxy.py", line 370, in run
    context.CLIARGS['func']()
  File "/home/adrian/src/ansible/lib/ansible/cli/galaxy.py", line 1006, in execute_publish
    publish_collection(collection_path, self.api, wait, timeout)
  File "/home/adrian/src/ansible/lib/ansible/galaxy/collection.py", line 383, in publish_collection
    import_uri = api.publish_collection(collection_path)
  File "/home/adrian/src/ansible/lib/ansible/galaxy/api.py", line 82, in wrapped
    return method(self, *args, **kwargs)
  File "/home/adrian/src/ansible/lib/ansible/galaxy/api.py", line 438, in publish_collection
    % (self.name, self.api_server))
  File "/home/adrian/src/ansible/lib/ansible/galaxy/api.py", line 183, in _call_galaxy
    raise GalaxyError(e, error_context_msg)
  File "/home/adrian/src/ansible/lib/ansible/galaxy/api.py", line 111, in __init__
    errors = err_info.get('errors', [])
AttributeError: 'list' object has no attribute 'get'
```

**Comment by alikins** [comment]
Fixed in ansible/galaxy-api#103
